### PR TITLE
Picasso bonding

### PR DESCRIPTION
**Pull request by easteregg**
## Issue
https://app.clickup.com/t/29xnna4

## Description
Picasso Bonding ➝ Bond PICA-KSM to purchase CHAOS at a discount 

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

File: frontend/ui-picasso/app/stores/defi/polkadot/bonds/slice.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { BondOffer } from "@/stores/defi/polkadot/bonds/types";
```
Comment (saadjhk) [inline frontend/ui-picasso/app/stores/defi/polkadot/bonds/slice.ts:1]: would prefer zustand store

**Review by saadjhk — CHANGES_REQUESTED** [review]

File: frontend/ui-picasso/app/components/Organisms/Bond/utils.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { Token } from "@/defi/Tokens";
+import { BondOffer } from "@/stores/defi/polkadot/bonds/types";
+import BigNumber from "bignumber.js";
+import { fromPica } from "@/defi/polkadot/pallets/BondedFinance";
+
+export function lpToSymbolPair(acc: string, token: Token) {
+  return acc.length > 0 ? acc + "-" + token.symbol : token.symbol;
```
[the remark this answers is not in the document]
Comment (easteregg) [reply frontend/ui-picasso/app/components/Organisms/Bond/utils.ts:7]: bond only have 1 assetId. This is for those asset ids that are representing a lp token.

File: frontend/ui-picasso/app/stores/defi/polkadot/bonds/slice.ts (hunk 1 above)
Comment (easteregg) [reply frontend/ui-picasso/app/stores/defi/polkadot/bonds/slice.ts:1]: Once the other PR is ready, we can refactor this.

File: frontend/ui-picasso/app/pages/bonds/[bond].tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+const standardPageSize = {
+  xs: 12,
+};
+
+type BondOfferBalances = {
+  [key: string]: BigNumber;
+};
+
+function useBalanceForOffer(offer: BondOffer) {
```
[the remark this answers is not in the document]
Comment (easteregg) [reply frontend/ui-picasso/app/pages/bonds/[bond].tsx:33]: offer has 2 assets internally. This is actually using that but adding 1 layer of abstraction.

File: frontend/ui-picasso/app/defi/polkadot/pallets/Oracle.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      })
+    );
+  });
+
+  return unsub;
+}
+
+export async function fetchAssetPrice(assetId: u128, api: ApiPromise) {
+  try {
+    const prices: any = await api.query.oracle.prices(assetId); // TODO[type-gen]: replace any with proper type
```
[the remark this answers is not in the document]
Comment (easteregg) [reply frontend/ui-picasso/app/defi/polkadot/pallets/Oracle.ts:31]: Not yet. But it is a good idea to subscribe this. will do sir!

File: frontend/ui-picasso/app/defi/polkadot/hooks/useBlockInterval.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+import { usePicassoProvider } from "@/defi/polkadot/hooks/index";
+
+// Some chains incorrectly use these, i.e. it is set to values such as 0 or even 2
+// Use a low minimum validity threshold to check these against
+const THRESHOLD = BN_THOUSAND.div(BN_TWO);
+const DEFAULT_TIME = new BN(6_000);
+
+const A_DAY = new BN(24 * 60 * 60 * 1000);
+
+function calcInterval(api: ApiPromise | undefined): BN | undefined {
```
[the remark this answers is not in the document]
Comment (saadjhk) [reply frontend/ui-picasso/app/defi/polkadot/hooks/useBlockInterval.ts:21]: makes sense, i just want to enforce use of BigNumber.js in our repos. having two different libraries for the same functionality is not a good idea

**Review by saadjhk — APPROVED** [review]

**Review by palankid — APPROVED** [review]

**Pull request merged**